CNF-22825: Consolidate around klog/v2

sebrandon1 opened the pull request:
## Summary

- Remove the vendored `go-logr/logr/testr` sub-package by replacing `testr.New(t)` with `klog.NewKlogr().WithName(ControllerName)` in both istiocsr and trustmanager test utilities
- Drop the `testing.T` parameter from `testReconciler()` in both packages and update all call sites
- Replace synthetic `fmt.Errorf()` + `log.Error()` calls with structured `log.V(N).Info()` at appropriate verbosity levels (rbacs.go, controller.go)
- Convert printf-style log format strings to structured key-value pairs (utils.go)

## Motivation

The `go-logr/logr/testr` package was only used in test helpers to create a logger backed by `testing.T`. Since the operator already depends on `klog/v2`, switching to `klog.NewKlogr()` removes an unnecessary vendored dependency without changing production behavior. The log-level fixes address cases where `log.Error()` was called with a fabricated error solely to log an informational message, and where format-string placeholders were used instead of structured key-value arguments.

Part of a broader effort to standardize on klog/v2 across repos:
- [rh-ecosystem-edge/eco-goinfra#1145](https://github.com/rh-ecosystem-edge/eco-goinfra/pull/1145)
- [redhat-best-practices-for-k8s/certsuite-qe#1215](https://github.com/redhat-best-practices-for-k8s/certsuite-qe/pull/1215)
- [redhat-best-practices-for-k8s/certsuite-qe#1216](https://github.com/redhat-best-practices-for-k8s/certsuite-qe/pull/1216)

## Jira

- [CNF-22825](https://issues.redhat.com/browse/CNF-22825) -- Consolidate istiocsr logging around klog/v2 (Code Review)

## Test Plan

- [ ] `make lint` passes
- [ ] Unit tests pass (`make test`) -- all test files updated to use the new `testReconciler()` signature
- [ ] Verify no remaining references to `go-logr/logr/testr` in vendor or source

sebrandon1 commented:
/retest ci/prow/e2e-operator

sebrandon1 commented:
/retest

sebrandon1 commented:
@coderabbitai resume

sebrandon1 commented:
Rebased onto current master (was 9 months stale). Cherry-picked the klog consolidation commit cleanly. All tests pass with 84.6% coverage on istiocsr package.

anandkuma77 commented:
remove 	"github.com/go-logr/logr" dependency from go.mod

anandkuma77 commented on pkg/controller/istiocsr/controller.go:
```diff
@@ -60,15 +58,14 @@ func New(mgr ctrl.Manager) (*Reconciler, error) {
 		CtrlClient:    c,
 		ctx:           context.Background(),
 		eventRecorder: mgr.GetEventRecorderFor(ControllerName),
-		log:           ctrl.Log.WithName(ControllerName),
```
Lost controller context,e old ctrl.Log.WithName(ControllerName) automatically tagged every log line with the controller name, making it easy to filter logs in a multi-controller operator. Bare klog.InfoS() calls are now indistinguishable from other controllers.

Why not use:
 klog.NewKlogr().WithName(ControllerName)

sebrandon1 committed:
Consolidate around klog/v2

Switch the istiocsr controller logger from ctrl.Log to a klog-backed logr.Logger
that preserves the controller name for filtering in multi-controller operators.
Remove the logr/testr test dependency in favor of klog.NewKlogr().

Replace synthetic fmt.Errorf() calls used solely to satisfy Error() with
appropriately leveled Info calls, and fix printf-style log messages to use
structured key-value pairs.

sebrandon1 commented:
Thanks for the review, @anandkuma77! Good catch on the lost controller context.

I've updated the PR to address your feedback:

1. **Restored the controller-scoped logger** — the `log logr.Logger` field is back on the `Reconciler` struct, initialized with `ctrl.Log.WithName(ControllerName)` (which is already klog-backed via `ctrl.SetLogger(klog.NewKlogr())` at startup). All log lines now carry the controller name again for filtering. This also stays consistent with how the trustmanager controller initializes its logger.

2. **Replaced synthetic `fmt.Errorf()` calls** — the 3 places that created throwaway errors solely to satisfy `ErrorS`/`Error` signatures (`"invalid label format"`, `"error updating clusterrole name in status"`, `"error updating clusterrolebinding name in status"`) now use appropriately leveled `r.log.V(n).Info()` calls instead, since these are validation/status conditions rather than operational errors.

3. **Fixed printf-style log message** — `utils.go` had a `r.log.V(4).Info("%s/%s istiocsr resource...")` using printf formatting; converted to structured key-value pairs.

The test logger was switched from `testr.New(t)` to `klog.NewKlogr().WithName(ControllerName)` to drop the `logr/testr` dependency per the original PR intent.

Squashed down to a single commit. PTAL when you get a chance.

sebrandon1 commented:
/retest ci/prow/e2e-operator-tech-preview

anandkuma77 commented:
LGTM,

sebrandon1 commented:
/retest

sebrandon1 commented:
/retest ci/prow/e2e-operator-tech-preview

sebrandon1 commented:
/rebase

sebrandon1 commented:
/retest

sebrandon1 commented:
/retest
